### ISSUE TITLE
cDNA dataset positive and negative issues

### ISSUE THREAD

**Issue by jiaweiguan** [jozhang97/MutateEverything#4]
Is the positive ddg value in the dataset a good mutation or a bad mutation?

**Comment by hippolytej** [comment]
Looks like the cDNA dataset from [Tsuboyama, K., Dauparas, J., Chen, J. et al. Mega-scale experimental analysis of protein folding stability in biology and design. Nature 620, 434–444 (2023)](https://doi.org/10.1038/s41586-023-06328-6) uses the definition DDG = DGwt - DGmut, whereby a positive DDG value means a stabilising (good) mutation, however in this implementation, the definition used is DDG = DGmut - DGwt, so the opposite basically - but let's wait for the authors to confirm this :)

**Comment by jozhang97** [comment]
sorry for the delay and thanks for your interest! In general, we try to use DDG = DGmut - DGwt.

but since not all works follow the same definition and we borrow data from previous works, I recommend the following sanity check that I commonly do: look at the dataset ddG distribution. Since most mutations are destabilizing, whatever side has larger mass is destabilizing. (mean works as well)

**Comment by MaxLiu245** [comment]
> sorry for the delay and thanks for your interest! In general, we try to use DDG = DGmut - DGwt.很抱歉延迟，感谢您的关注！一般来说，我们尝试使用DDG = DGmut - DGwt。
> 
> but since not all works follow the same definition and we borrow data from previous works, I recommend the following sanity check that I commonly do: look at the dataset ddG distribution. Since most mutations are destabilizing, whatever side has larger mass is destabilizing. (mean works as well)但由于并非所有作品都遵循相同的定义，并且我们从以前的作品中借用数据，我建议进行以下我经常做的健全性检查：查看数据集ddG分布。因为大多数突变都是不稳定的，所以无论哪一方具有更大的质量都是不稳定的。(mean工作也）

Hi, I had a question on how you calculated the label DDG because I got a unmatched DDG result from yours if I began from cDNA dataset. Hope you can help me.

I want to use this example, i.e. the first piece of data W1Q for 1GYZ in your `mutate_everything_data\cdna\mutations\cdna_train.csv`. Here the DDG is about -0.22877.
![image](https://github.com/user-attachments/assets/f4ec93c0-165e-47f7-81a4-318b9b0e0cf4)

Then I downloaded cDNA dataset (Version v2_230420) from https://zenodo.org/records/7992926, where I fetched the data for `1GYZ.pdb` and  `1GYZ.pdb_W1Q` from `Processed_K50_dG_datasets\Processed_K50_dG_datasets\Tsuboyama2023_Dataset2_Dataset3_20230416.csv` as follows.
![image](https://github.com/user-attachments/assets/ab3c2c30-9e6b-4b25-bb23-1d45fb346d7a)

Here I calculated all the differences between `1GYZ.pdb` and  `1GYZ.pdb_W1Q`, but didn't find any -0.22877.

Could you kindly tell us how to get the right DDG from cDNA dataset? I felt really confused here. Looking forward to you!

**Comment by jozhang97** [comment]
Hi, 

thanks for your interest

can you check if the negative (+0.22) is in the dataset?

If not, could you share your data preprocessing script and output files so I can check?

**Comment by MaxLiu245** [comment]
Hi, I didn't write any scripts but only (want to) saw and checked the **labels** (i.e. DDG) of MutateEverything.

What I have done are:

1. Downloaded the cDNA dataset from https://drive.google.com/drive/folders/1psp5LBnAWpwkzGtsWD9SJgauo8xX_eaK?usp=sharing (in README).
2. Opened file `cdna_train.csv` in fold `mutate_everything_data\cdna\mutations`, and the first piece of data was as below. Here muation W1Q of 1gyz has a DDG of -0.22877.

Unnamed: 0 | pdb_id | ddg | mut_info | mut_seq | fr1 | pos1 | to1 | mut_info1 | mut_info2 | fr2 | pos2 | to2 | wt_seq
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
0 | 1gyz | -0.22877 | W1Q | QIARINAAVRAYGLNYSTFINGLKKAGIELDRKILADMAVRDPQAFEQVVNKVKEALQV | W | 1 | Q |   |   |   |   |   | WIARINAAVRAYGLNYSTFINGLKKAGIELDRKILADMAVRDPQAFEQVVNKVKEALQV

3. Then I wanted to see how you got this DDG from original data, so I came to check the paper mentioned by @hippolytej .
4. So I Downloaded this paper's data, i.e. cDNA dataset (Version v2_230420) from https://zenodo.org/records/7992926.
5. The following step was as what I said before. I searched and found the records of (1) `W1Q` of `1gyz` and (2) `1gyz wide type`.

Till now I find that the difference between each column of `W1Q` and `WT` is not -0.22877. This is why I have on idea on how you calculated this DDG. Hope you can help me.

**Comment by jozhang97** [comment]
Hi,

Thanks for your diligence. I followed your steps. To add some more details, in step 4, I downloaded `Processed_K50_dG_datasets.zip` from the link you shared and used `K50_dG_Dataset1_Dataset2.csv`.

One entry in this table is (filtering columns):
name | dG_ML | ddG_ML
|------|-------|--------|
1GYZ.pdb_W1Q | 4.237097399066746 | 0.2287745574547584

We used the ddG_ML that they calculated, which matches with the one in our released training set (up to a sign)

**Comment by MaxLiu245** [comment]
Hi,

Thanks for you updating. I just noticed that the data you used is [Version v1](https://zenodo.org/records/7401275) released on 2022-12, while mine (before) was of Version v2_230420 released on 2023-04.

There's the change between the two versions with `1GYZ.pdb_W1Q`, which led to my past unmatched observation (really interesting if the attributes change). Again thanks for your time!

**Comment by jozhang97** [comment]
I see, so the versions have some differences as well. the ddG_ML from above is also from the latest version (2023-04) but there seem to be some discrepancies between the versions

Ill close this for now, feel free to reopen if there are any lingering issues.
